plugin/src/test/java/org/opensearch/ml/rest/SecureMLRestIT.java

```diff
@@ -142,4 +144,54 @@ public void testTrainAndPredictWithFullAccess() throws IOException {
             }
         );
     }
+
+    public void testTrainModelWithFullAccessThenPredict() throws IOException {
+        KMeansParams kMeansParams = KMeansParams.builder().build();
+        // train model
+        train(mlFullAccessClient, FunctionName.KMEANS, irisIndex, kMeansParams, searchSourceBuilder, trainResult -> {
+            String modelId = (String) trainResult.get("model_id");
+            assertNotNull(modelId);
+            String status = (String) trainResult.get("status");
+            assertEquals(MLTaskState.COMPLETED.name(), status);
+            try {
+                getModel(mlFullAccessClient, modelId, model -> {
+                    String algorithm = (String) model.get("algorithm");
+                    assertEquals(FunctionName.KMEANS.name(), algorithm);
+                });
+            } catch (IOException e) {
+                assertNull(e);
+            }
+            try {
+                // predict with trained model
+                predict(mlFullAccessClient, FunctionName.KMEANS, modelId, irisIndex, kMeansParams, searchSourceBuilder, predictResult -> {
+                    String predictStatus = (String) predictResult.get("status");
+                    assertEquals(MLTaskState.COMPLETED.name(), predictStatus);
+                    Map<String, Object> predictionResult = (Map<String, Object>) predictResult.get("prediction_result");
+                    ArrayList rows = (ArrayList) predictionResult.get("rows");
+                    assertTrue(rows.size() > 1);
```

Comment:
Can there be any more specific check than result just being existent?

Reply:
Will be hard to check, the algorithm has some built-in randomness, the result could vary. Prefer to keep the verification simple here to avoid flaky tests. And here our main goal is to test the security part. Should be ok to verify there is result generated which means the user has permission and can run the action successfully.

Reply:
Okay that makes sense not to do it here and now, maybe in the future some tests can be added similar to AD with some sort of minimum precision in a different place and maybe for other algorithms (will have to approach each algo differently), but also as we know such tests is asking for more flakiness 😆

Reply:
Yes, we should avoid over-engineering, that may bring unexpected extra load to us.